### ISSUE TITLE
Array slice bug: last element duplication

### ISSUE THREAD

**Issue by MeloJR** [godotengine/godot#38969]
**Godot version:**
v3.2.1

**OS/device including version:**
Windows 10

**Issue description:**
If I call the `slice` method with the `end` value greater than the array size it duplicates the last element unless the array size was 1.

**Steps to reproduce:**
```
[1,2,3].slice(0, 3) // = [1,2,3,3]
[1,2].slice(0, 3) // = [1,2,2]
```

**Comment by akien-mga** [comment]
Probably fixed in 3.2.2 beta 3, can you test?  https://downloads.tuxfamily.org/godotengine/3.2.2/beta3/

**Comment by madmiraal** [comment]
Duplicate of #34284, and I've confirmed that it's fixed in 3.2.2 beta 3.

**Comment by MeloJR** [comment]
I'm sorry for the duplicate, I only searched in open issues 😅. Thank you for the answer.
